internal/core/application/operator_service_internal_test.go

```diff
@@ -0,0 +1,47 @@
+package application
+
+import (
+	"testing"
+	"time"
+)
+
+func TestInitGroupedVolume(t *testing.T) {
+	start := time.Now()
+	type args struct {
+		start            time.Time
+		end              time.Time
+		groupByTimeFrame int
+	}
+	tests := []struct {
+		name string
+		args args
+		want int
+	}{
+		{
+			name: "1",
+			args: args{
+				start:            start,
+				end:              start.Add(12 * time.Hour),
+				groupByTimeFrame: 4,
+			},
+			want: 3,
+		},
+		{
+			name: "2",
+			args: args{
+				start:            start,
+				end:              start.Add(12 * time.Hour),
+				groupByTimeFrame: 1,
+			},
+			want: 12,
+		},
+	}
```

Comment:
I think it would be really valuable to have a series of test cases here with `start`, `end` assuming values reflecting the `TimeRange` proto enum entries and for each of these ranges, `groupByTimeFrame` assuming all the values reflecting `TimeFrame` proto enum.

I guess some of them should be invalid: in general `groupByTimeFrame` shouldn't be <= `TimeRange.EndTime - TimeRange.StartTime`?

I mean, it would not have that much sense for example a timerange of a week, with a timeframe of a month or a year, do you agree?

Reply:
timeframe < timerange ofc

if timeframe == timerange you would see a single candle in the chart so to speak, not that useful, but valid

Reply:
Done